libs/sdk-core/src/breez_services.rs

```diff
@@ -1348,25 +1340,22 @@ fn get_cheapest_opening_fee_params(
 }
 
 fn get_longest_valid_opening_fee_params(
-    opening_fee_params_menu: &mut Vec<models::OpeningFeeParams>,
-) -> Option<models::OpeningFeeParams> {
-    match validate_opening_fee_params_menu(opening_fee_params_menu) {
-        Ok(()) => {
-            // TODO: We need to pick cheapest opening fee params that's valid for at least 48 hours
-            opening_fee_params_menu.sort_by(|a, b| {
-                let dt1 =
-                    DateTime::parse_from_str(&a.valid_until, "%Y %b %d %H:%M:%S%.3f %z").unwrap();
-                let dt2 =
-                    DateTime::parse_from_str(&b.valid_until, "%Y %b %d %H:%M:%S%.3f %z").unwrap();
-                dt2.cmp(&dt1)
-            });
-            Some(opening_fee_params_menu[0].clone())
-        }
-        Err(e) => {
-            error!("failed to validate opening_fee_params_menu: {e}");
-            None
-        }
-    }
+    opening_fee_params_menu: &Vec<OpeningFeeParams>,
+) -> Result<Option<OpeningFeeParams>> {
+    validate_opening_fee_params_menu(opening_fee_params_menu)?;
+
+    // Find the fee params that are valid for at least 48h
+    let now = Utc::now();
+    let duration_48h = chrono::Duration::hours(48);
+    let mut valid_min_48h: Vec<OpeningFeeParams> = opening_fee_params_menu
+        .iter()
+        .filter(|f| f.valid_until - now > duration_48h)
+        .cloned()
+        .collect();
+
+    // Of those, return the cheapest
+    valid_min_48h.sort_by_key(|f| f.min_msat);
```

Comment:
Note that these have to be sorted already according to the spec. So if the menu is valid, you shouldn't have to sort here again. Also, there could be multiple params with the same `min_msat`, but a different `proportional`.